### PR TITLE
Track whether the RCP hang timer has started to prevent double setting

### LINKED ISSUE #679: OSINSERTTIMER Crash
image

Reproduceable on Parallel by enabling highest-demanding `BETTER_REVERB` settings and running around the castle for a bit. My guess is this is an issue with the audio thread somehow clashing with the scheduler, but regardless this should not be happening. Unproven whether this is emulator only.

Marking this as high priority until this is both confirmed to be an unlikely issue on hardware and also confirmed to be unrelated to performance issues when present on the game thread.

### PR DESCRIPTION
**Pull request by someone2639**
Closes #679 once tested more thoroughly

### CODE REVIEW

File: src/boot/main.c [hunk 1]
```diff
@@ -303,13 +303,19 @@ void handle_dp_complete(void) {
...
+        osSetTimer(&RCPHangTimer.timer, OS_USEC_TO_CYCLES(3000000), (OSTime) 0, &gIntrMesgQueue, (OSMesg) MESG_RCP_HUNG);
+        RCPHangTimer.started = TRUE;
+    }
 }
 
+void osStopTimer_log(OSTimer *);
 void stop_rcp_hang_timer(void) {
-    osStopTimer(&RCPHangTimer);
+    osStopTimer(&RCPHangTimer.timer);
+    RCPHangTimer.started = FALSE;
```
Comment (gheskett) [inline src/boot/main.c:318]: Does this also need a conditional check or does it not matter?

File: src/boot/main.c [hunk 2]
```diff
@@ -303,13 +303,19 @@ void handle_dp_complete(void) {
     sCurrentDisplaySPTask = NULL;
 }
 
-OSTimer RCPHangTimer;
+OSTimerEx RCPHangTimer;
+int osSetTimer_log(OSTimer* t, OSTime countdown, OSTime interval, OSMesgQueue* mq, OSMesg msg);
```
Comment (gheskett) [inline src/boot/main.c:307]: Remove this line

**Review by gheskett — CHANGES_REQUESTED** [review]

File: src/boot/main.c [hunk 3]
```diff
@@ -303,13 +303,19 @@ void handle_dp_complete(void) {
...
 void start_rcp_hang_timer(void) {
-    osSetTimer(&RCPHangTimer, OS_USEC_TO_CYCLES(3000000), (OSTime) 0, &gIntrMesgQueue, (OSMesg) MESG_RCP_HUNG);
+    if (RCPHangTimer.started == FALSE) {
+        osSetTimer(&RCPHangTimer.timer, OS_USEC_TO_CYCLES(3000000), (OSTime) 0, &gIntrMesgQueue, (OSMesg) MESG_RCP_HUNG);
+        RCPHangTimer.started = TRUE;
+    }
 }
 
+void osStopTimer_log(OSTimer *);
```
Comment (gheskett) [inline src/boot/main.c:315]: And this one

File: src/boot/main.c (hunk 1 above)
Comment (someone2639) [reply src/boot/main.c:318]: Stopping a timer twice seems to work just fine, it's starting one twice that breaks things

File: src/boot/main.c (hunk 2 above)
Comment (someone2639) [reply src/boot/main.c]: addressed

**Review by gheskett — APPROVED** [review]

**Review by thecozies — APPROVED** [review]

**Pull request merged**